lib/commands/alert.js

```diff
@@ -94,16 +94,33 @@ commands.mobileHandleAlert = async function (opts = {}) {
 };
 
 helpers.getAlert = async function () {
-  let possibleAlert = await this.findNativeElementOrElements('class name', 'XCUIElementTypeScrollView', true);
+  // ScrollView should always be found, unless Safari isn't even running.
+  let possibleAlert = await this.findNativeElementOrElements(
+    'class name', 'XCUIElementTypeScrollView', true
+  );
   if (possibleAlert.length !== 1) {
     throw new errors.NoAlertOpenError();
   }
-
-  let possibleAlertButtons = await this.findNativeElementOrElements('class name', 'XCUIElementTypeButton', true, possibleAlert[0].ELEMENT);
-  if (possibleAlertButtons.length  < 1 || possibleAlertButtons.length > 2) {
+  let response;
+  // We expect exactly one TextView if an alert is present.
+  response = await this.findNativeElementOrElements(
```

Comment:
Do we really need this addition?
How confident are you that there should always be one XCUIElementTypeTextView .
Presumably modal dialogs can be implemented in various ways, and some may not have that element?

Reply:
I had assumed that a Safari modal dialog would always have the same structure.  That's certainly the case for our elements/modal dialog UJs.  I'll do some more testing.  

E.g. there is a "prompt" style modal, which you can enter text into - though we don't support those in the engine.  I assumed that would also contain a `TextView`, though, for the static part of the dialog.

I suppose we don't strictly need the `TextView` check.  It helps in the (admittedly rare) case of a web page with one or two buttons.  In that case, Appium would have to get the full `Button` elements to check the text inside them (for "ok" and "cancel").

Reply:
I've updated the modal dialog UJs to test the `confirm` and `prompt` style dialogs.  They are both detected and dismissed fine with these changes.

Reply:
When I said tests I meant the unit tests.